tests/30rooms/60anonymousaccess.pl

```diff
@@ -189,6 +209,46 @@ sub await_event_not_presence_for
       });
    };
 
+test "Annonymous user can call /sync on a world readable room",
+   requires => [ anonymous_user_fixture(), local_user_fixture() ],
+
+   do => sub {
+      my ( $anonymous_user, $user ) = @_;
+
+      my ($room_id, $sent_event_id);
+
+      matrix_create_and_join_room( [ $user ] )
+      ->then( sub {
+         ( $room_id ) = @_;
+
+         matrix_set_room_history_visibility( $user, $room_id, "world_readable" );
+      })->then( sub {
+         matrix_send_room_text_message( $user, $room_id, body => "mice" );
+      })->then( sub {
+         ( $sent_event_id ) = @_;
+
+         matrix_sync( $anonymous_user, filter => encode_json({
+            room => {
+               rooms => [ $room_id ],
+               ephemeral => { types => [] },
+               state => { types => [] },
+               timeline => { types => ["m.room.message"] },
+            },
+            presence => { types => [] }
+         }));
+      })->then( sub {
+         my ( $sync_body ) = @_;
+
+         assert_eq(
+            $sent_event_id,
+            $sync_body->{rooms}{join}{$room_id}{timeline}{events}[0]{event_id},
```

Comment:
This is quite a deep search. I'd be tempted to break it up a little so if it fails, you get some idea why:

    assert_json_object( my $room = $sync_body->{rooms}{join}{$room_id} );
    assert_json_list( my $events = $room->{timeline}{events} );
    assert_eq( $events->[0]{event_id}, $sent_event_id, 'event id' );

It's pretty arbitrary but if it fails the line number might help determine which bit